streams/src/test/java/org/apache/kafka/streams/kstream/internals/SuppressScenarioTest.java

```diff
@@ -517,6 +517,180 @@ public void shouldSupportFinalResultsForSessionWindows() {
         }
     }
 
+    @Test
+    public void shouldWorkBeforeGroupBy() {
```

Comment:
Added tests verifying the thing actually works in the operators that depend on it.